src/app/core/entity/entity-mapper.service.spec.ts

```diff
@@ -172,4 +172,59 @@ describe("EntityMapperService", () => {
     expect(loadedByFullId._id).toBe(loadedByEntityId._id);
     expect(loadedByFullId._rev).toBe(loadedByEntityId._rev);
   });
+
+  it("publishes updates to any listeners", (done) => {
+    const testId = "t1";
+    receiveUpdatesAndTestTypeAndId(undefined, testId, done);
+
+    const testEntity = new Entity(testId);
+    entityMapper.save(testEntity, true);
+    entityMapper.remove(testEntity);
+  });
+
+  it("publishes when an existing entity is updated", async (done) => {
+    receiveUpdatesAndTestTypeAndId("update", existingEntity.entityId, done);
+
+    const loadedEntity = await entityMapper.load<Entity>(
+      Entity,
+      existingEntity.entityId
+    );
+    await entityMapper.save<Entity>(loadedEntity);
+  });
+
+  it("publishes when an existing entity is deleted", async (done) => {
+    receiveUpdatesAndTestTypeAndId("remove", existingEntity.entityId, done);
+
+    const loadedEntity = await entityMapper.load<Entity>(
+      Entity,
+      existingEntity.entityId
+    );
+    await entityMapper.remove<Entity>(loadedEntity);
+  });
+
+  it("publishes when a new entity is being saved", (done) => {
+    const testId = "t1";
+    receiveUpdatesAndTestTypeAndId("new", testId, done);
+
+    const testEntity = new Entity(testId);
+    entityMapper.save(testEntity, true);
+  });
+
+  function receiveUpdatesAndTestTypeAndId(
+    type: string | undefined,
+    entityId: string | undefined,
```

Comment:
If it's undefined, the check won't happen (it will only test the defined values). That way one can test only the type (or id) with this one function. Will change to optional!